app/src/features/charts/TableChart.tsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { Paper } from "@material-ui/core";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableContainer from "@material-ui/core/TableContainer";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import { Row } from "../../utils/DatasetTypes";
+
+function TableChart(props: { data: Row[]; columns?: string[] }) {
+  const tableColumns: string[] =
+    props.columns === undefined ? Object.keys(props.data[0]) : props.columns;
+
+  return (
+    <TableContainer component={Paper}>
+      <Table>
+        <TableHead>
+          <TableRow>
+            {tableColumns.map((columnName) => (
+              <TableCell>{columnName}</TableCell>
+            ))}
+          </TableRow>
```

Comment:
Might be nice to style the header row slightly differently so it's more clear it's the header row